go-controller/pkg/clustermanager/status_manager/egressqos_manager.go

```diff
@@ -0,0 +1,79 @@
+package status_manager
+
+import (
+	"context"
+	"strings"
+
+	egressqosapi "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/egressqos/v1"
+	egressqosapply "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/egressqos/v1/apis/applyconfiguration/egressqos/v1"
+	egressqosclientset "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/egressqos/v1/apis/clientset/versioned"
+	egressqoslisters "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/egressqos/v1/apis/listers/egressqos/v1"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type egressQoSManager struct {
+	lister egressqoslisters.EgressQoSLister
+	client egressqosclientset.Interface
+}
+
+func newEgressQoSManager(lister egressqoslisters.EgressQoSLister, client egressqosclientset.Interface) *egressQoSManager {
+	return &egressQoSManager{
+		lister: lister,
+		client: client,
+	}
+}
+
+//lint:ignore U1000 generic interfaces throw false-positives https://github.com/dominikh/go-tools/issues/1440
+func (m *egressQoSManager) get(namespace, name string) (*egressqosapi.EgressQoS, error) {
+	return m.lister.EgressQoSes(namespace).Get(name)
+}
+
+//lint:ignore U1000 generic interfaces throw false-positives
+func (m *egressQoSManager) getMessages(egressQoS *egressqosapi.EgressQoS) []string {
+	return egressQoS.Status.Messages
+}
+
+//lint:ignore U1000 generic interfaces throw false-positives
+func (m *egressQoSManager) updateStatus(egressQoS *egressqosapi.EgressQoS, applyOpts *metav1.ApplyOptions,
+	applyEmptyOrFailed bool) error {
+	if egressQoS == nil {
+		return nil
+	}
+	newStatus := "EgressQoS Rules applied"
+	for _, message := range egressQoS.Status.Messages {
+		if strings.Contains(message, types.EgressQoSErrorMsg) {
+			newStatus = types.EgressQoSErrorMsg
+			break
+		}
+	}
+	if applyEmptyOrFailed && newStatus != types.EgressQoSErrorMsg {
+		newStatus = ""
```

Comment:
when can this happen that we want to "clear" a status and set it to empty?
feels weird right? either you update to a +v1 value or an error/-ve value.
what am I missing..

Reply:
This makes `status.status`  to be empty until all ic zone processed EgressQoS object successfully and then update status with 'success'. The variable `applyEmptyOrFailed` helps to do that.

Reply:
ahh I see.